src/Umbraco.Core/Security/MembershipProviderBase.cs

```diff
@@ -520,7 +521,9 @@ public override string GetPassword(string username, string answer)
 
         public override string ResetPassword(string username, string answer)
         {
-            if (EnablePasswordReset == false)
+            var auth = new HttpContextWrapper(HttpContext.Current).GetUmbracoAuthTicket();
+            var userIsAdmin = ApplicationContext.Current.Services.UserService.GetByUsername(auth.Name).IsAdmin();

```

Comment:
There is no null check here, if `GetByUsername` fails you'll get a null reference exception